Error "The output from `csharp-models-to-json` contains invalid JSON." on nuget warnings in projects

Version: "csharp-models-to-typescript": "^0.21.1"

After recent Visual Studio 2022 update I started to get NU1803 warnings compiling projects (actually not compiling, but restoring). And `npm run generate-types` stopped working.

`generate-types` defined in package.json scripts:
```
"generate-types": "csharp-models-to-typescript --config=cs-to-ts-config.json"
```

Output from `npm run generate-types`:
```
> csharp-models-to-typescript --config=cs-to-ts-config.json

The output from `csharp-models-to-json` contains invalid JSON.

Unexpected token C in JSON at position 0

C:\sources\ReactApp\ClientApp\node_modules\csharp-models-to-typescript\lib\csharp-models-to-json\csharp-models-to-json.csproj : warning NU1803: You are running the 'restore' operation with an 'HTTP' source, 'http://nuget.mycompany.com/v3/index.json'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source.
C:\sources\ReactApp\ClientApp\node_modules\csharp-models-to-typescript\lib\csharp-models-to-json\csharp-models-to-json.csproj : warning NU1803: You are running the 'restore' operation with an 'HTTP' source, 'http://nuget.mycompany.com/v3/index.json'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source.
[{"FileName":"C:\\sources\\ReactApp\Backend\\Models\\Customer.cs","Models":[{"ModelName":
...
```

Proper JSON goes after warnings.

The quick fix is to change line in [index.js](https://github.com/svenheden/csharp-models-to-typescript/blob/master/index.js) to remove warning lines:
```js
json = JSON.parse(stdout);
```

to this:
```js
const removeWarningLines = (str) =>
    str.includes("csharp-models-to-json.csproj : warning") ? "" : str;

stdout = stdout.split("\n").map(removeWarningLines).join("\n");
stdout = stdout.split("\r").map(removeWarningLines).join("\r");
json = JSON.parse(stdout);
```

Update to this issue.  We're also starting to see a similar thing but with a different set of warnings:
```
The output from 'csharp-models-to-json' contains invalid JSON.

warning NETSDK1138: The target framework 'netcoreapp3.1' is out of support and will not receive security updates in the future. Please refer to https://aka.ms/dotnet-core-support for more information about the support policy.

warning NU1903: Package 'Newtonsoft.Json' 12.0.3 has a known high severity vulnerability, https://github.com/advisories/GHSA-5crp-9r3c-p9vr
```
For new team members, installing .net core 3.1 runtime does not fix this issue.  I had to recompile the csharp-models-to-json project using .net 7 after updating the target platform and updating the nuget packages.  Then I zipped that up to distribute to the team and am having team members replace the csharp-models-to-json folder in their node modules with the new one.  

I can confirm this does fix it.

Obviously, this is less than ideal as any reinstall of the node modules will blow away this change.  Thankfully, this conversion process for us is not a part of ci, otherwise we'd probably have to look into forking the library.

I think it would be valuable to have this library updated to .net 7 at least.  Alternatively, you may be able to compile multiple versions (though this will admittedly increase the package size).

Just some thoughts for improvement as well as a sure-fire way for anyone else with this problem to solve it.  You could easily change the .net version to support whatever environment you are working with.

--- a/index.js
+++ b/index.js
@@ -61,7 +61,18 @@ dotnetProcess.stdout.on('end', () => {
     let json;
 
     try {
-        json = JSON.parse(stdout);
+        // Extract the JSON content between the markers
+        const startMarker = '<<<<<<START_JSON>>>>>>';
+        const endMarker = '<<<<<<END_JSON>>>>>>';
+        const startIndex = stdout.indexOf(startMarker);
+        const endIndex = stdout.indexOf(endMarker);
+
+        if (startIndex !== -1 && endIndex !== -1 && endIndex > startIndex) {
+            const jsonString = stdout.substring(startIndex + startMarker.length, endIndex).trim();
+            json = JSON.parse(jsonString);
+        } else {
+            throw new Error('JSON markers not found or invalid order of markers.');
+        }
     } catch (error) {
         return console.error([
             'The output from `csharp-models-to-json` contains invalid JSON.',

--- a/lib/csharp-models-to-json/Program.cs
+++ b/lib/csharp-models-to-json/Program.cs
@@ -5,6 +5,7 @@ using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.Extensions.Configuration;
 using Ganss.IO;
+using System.Text;
 
 namespace CSharpModelsToJson
 {
@@ -36,7 +37,13 @@ namespace CSharpModelsToJson
             }
 
             string json = JsonSerializer.Serialize(files);
-            System.Console.WriteLine(json);
+
+            var sb = new StringBuilder();
+            sb.AppendLine("<<<<<<START_JSON>>>>>>");
+            sb.AppendLine(json);
+            sb.AppendLine("<<<<<<END_JSON>>>>>>");
+
+            System.Console.WriteLine(sb.ToString());
         }
 
         static List<string> getFileNames(List<string> includes, List<string> excludes) {

